### PR TITLE
[compiler] More tests for "for" loops

### PR DESCRIPTION
**Pull request by brmataptos**
### Description
Add a few more tests for `for` loops.  Copy some tests over to move-compiler-v2 to also test there.
For correct code, use transactional-tests, but for compiler errors, use tests.

Also fix a typo in one test making it not useful.

Addresses Issue #11583.

### Test Plan
These are tests.

### CODE REVIEW

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move [hunk 1]
```diff
@@ -2,11 +2,13 @@
 script {
     fun main() {
         let x = 0;
-        for (i in 0..10) {
-            if (x >= 5) break;
-            x = x + 1;
-            continue
-        }
-        assert!(move x == 5, 42);
+        for (i in 0..10) { // 0,  1,  2,  3,  4, 5, 6, 7, 8, 9, 10
```
Comment (vineethk) [inline third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:5]:
```suggestion
        for (i in 0..10) { // 0,  1,  2,  3,  4, 5, 6, 7, 8, 9
```

Comment (vineethk) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:5]: Upper bound is exclusive.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
+            x = x + 1;     // 1,  4, 13, 16,
+            if (x >= 15) break;
+	    x = x + 2;     // 3,  6, 15,  -
```
Comment (vineethk) [inline third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:8]:
```suggestion
	        x = x + 2;     // 3,  6, 15,  -
```

Comment (vineethk) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:8]: Indentation here and below (assignments to `x`) should be fixed for readability.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.exp1 [hunk 3]
```diff
@@ -0,0 +1,151 @@
+processed 1 task
+
+task 0 'print-bytecode'. lines 1-25:
```
Comment (vineethk) [inline third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.exp1:3]: What is this `*.exp1` file?

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//# run
+script {
+    fun main() {
+        let k = 0;
+	let y = 0;
+	for (j in 0..10) { //j:  0,  1,  2,  3,  4, 5, 6, 7, 8, 9, 10
```
Comment (vineethk) [inline third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move:6]: The indentation in this test is broken: makes it hard to read.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/function_call_mut_arg_twice.move [hunk 5]
```diff
@@ -0,0 +1,18 @@
+//# publish
```
Comment (vineethk) [inline third_party/move/move-compiler/transactional-tests/tests/control_flow/function_call_mut_arg_twice.move:1]: Looks like many non-for-loop related tests (like this one) were also added. I am assuming this is intentional.

File: third_party/move/move-compiler-v2/tests/checking/control_flow/infinite_loop_with_dead_exits.move [hunk 6]
```diff
@@ -0,0 +1,18 @@
+address 0x42 {
+module M {
+
+    // fun t1() {
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/tests/checking/control_flow/infinite_loop_with_dead_exits.move:4]: Weird to have commented out code here. Any particular reason?

File: third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop.move [hunk 7]
```diff
@@ -0,0 +1,14 @@
+//# run
+script {
+    fun main() {
+        let x = 0;
+        for (i in 0..10) { // 0,  1,  2,  3,  4, 5, 6, 7, 8, 9, 10
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop.move:5]: Similar comments as made on the "original" file.

File: third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    fun main() {
+        let k = 0;
+	let y = 0;
+	for (j in 0..10) { //j:  0,  1,  2,  3,  4, 5, 6, 7, 8, 9, 10
+	    y = y + 1;     //y:  1,  4, 13, 16,
+	    if (y >= 15) break;
+	    y = y + 2;     //y:  3,  6, 15,  -
+	    let x = 0;
+            for (i in 0..10) { //i:  0,  1,  2,  3,  4, 5, 6, 7, 8, 9, 10
+		x = x + 1;     //x:  1,  4, 13, 16,
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move:12]: Please fix indentation if possible.

**Review by vineethk — COMMENTED** [review]
Have some minor comments, LGTM otherwise.

Thanks for doing this!

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move (hunk 1 above)
Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:5]: fxied.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move (hunk 2 above)
Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop.move:8]: oops, emacs added tabs.  gotta get rid of that.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.exp1 (hunk 3 above)
Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.exp1:3]: oops, was a backup file.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move (hunk 4 above)
Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move:6]: again, tab issue.

File: third_party/move/move-compiler/transactional-tests/tests/control_flow/function_call_mut_arg_twice.move (hunk 5 above)
Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/function_call_mut_arg_twice.move:1]: I was making /control_flow/ the same for V1 and V2.

Comment (brmataptos) [reply third_party/move/move-compiler/transactional-tests/tests/control_flow/function_call_mut_arg_twice.move:1]: But I've also backpedaled and removed a test that doesn't yield expected warnings on -v2 now.

File: third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move (hunk 8 above)
Comment (brmataptos) [reply third_party/move/move-compiler-v2/transactional-tests/tests/control_flow/break_continue_for_loop_nested.move:12]: Sorry, tabs again.

**Review by vineethk — APPROVED** [review]

**Review by wrwg — APPROVED** [review]

**Pull request merged**